crates/cast/src/starknet_commands/declare.rs

```diff
@@ -57,12 +56,14 @@ pub async fn declare(
     contract_name: &str,
     max_fee: Option<FieldElement>,
     account: &mut SingleOwnerAccount<&JsonRpcClient<HttpTransport>, LocalWallet>,
+    path_to_scarb_toml: &Option<Utf8PathBuf>,
 ) -> Result<DeclareResponse> {
     let contract_name: String = contract_name.to_string();
-    which::which("scarb")
-        .context("Cannot find `scarb` binary in PATH. Make sure you have Scarb installed https://github.com/software-mansion/scarb")?;
+    let manifest_path =
+        get_scarb_manifest(path_to_scarb_toml).expect("Failed to obtain scarb manifest file path");
     let command_result = Command::new("scarb")
```

Comment:
vide supra
```suggestion
        let mut cmd = Command::new("scarb");
        if let Some(manifest_path) = path_to_scarb_toml {
            cmd.arg("--manifest-path").arg(manifest_path);
        }
        let command_result = cmd.arg("--release").(...)
```

Reply:
Changed the way get_scarb_manifest works, so right now it is either a user defined path or found one using scarb, as per https://github.com/foundry-rs/starknet-foundry/pull/340#discussion_r1279390005.

Do you think it can be left as is? Either way the path exists, and my understanding is that scarb just won't look for it for the second time here, but will take the path directly right?

Reply:
Yes, if one present Scarb will used passed path verbatim.